### PR TITLE
add solid body rotation test

### LINKED ISSUE #164: a bug in energy equation
pressure work to the energy tendency is missing [here](https://github.com/CliMA/ClimaAtmos.jl/blob/main/src/Models/Nonhydrostatic3DModels/nonhydrostatic_3d_model.jl#L181-L184).

See https://github.com/CliMA/ClimaCore.jl/pull/411

### PR DESCRIPTION
**Pull request by jiahe23**
- fix the bug in energy equation (close #164);
- add regression and validation tests for the solid body rotation case.

### CODE REVIEW

**Comment by bischtob** [comment]
bors try

File: test/test_cases/run_solid_body_rotation.jl [hunk 1]
```diff
@@ -42,11 +43,33 @@ function run_solid_body_rotation(
...
+        set!(simulation, ρ = ρ, uh = uh, w = w, ρe_tot = ρe_tot)
+        run!(simulation)
+        u_end = simulation.integrator.u.nhm
+
+        uh_phy = Geometry.transform.(Ref(Geometry.UVAxis()), u_end.uₕ)
+        w_phy = Geometry.transform.(Ref(Geometry.WAxis()), u_end.w)
+
+        @test maximum(abs.(uh_phy.components.data.:1)) ≤ 1e-11
+        @test maximum(abs.(uh_phy.components.data.:2)) ≤ 1e-11
+        @test maximum(abs.(w_phy |> parent)) ≤ 1.0
```
Comment (bischtob) [inline test/test_cases/run_solid_body_rotation.jl:72]: Does this mean, we are satisfied with less than 1m/s on w? Seems large, no?

Comment (jiahe23) [reply test/test_cases/run_solid_body_rotation.jl:72]: Yes. It is large for the first several hours. It damps away for a longer run (~1e-3 after 1 simulation day).

Comment (jiahe23) [reply test/test_cases/run_solid_body_rotation.jl:72]: Initial conditions not in discrete balance is a potential cause of the flow.

**Comment by bischtob** [comment]
When the correctness of the validation test has been confirmed by @jiahe23 we can merge this PR.

**Comment by jiahe23** [comment]
bors try

**Review by bischtob — APPROVED** [review]
LGTM!

**Pull request merged**